services/arrow-flight/src/test/java/com/dremio/service/flight/AbstractTestFlightSqlServerCatalogMethods.java

```diff
@@ -0,0 +1,224 @@
+    }
+  }
+
+  @Test
+  public void testGetTablesFilteringByTablePattern() throws Exception {
+    FlightSqlClient flightSqlClient = getFlightClientWrapper().getSqlClient();
+    FlightInfo flightInfo = flightSqlClient.getTables(null, null, "COLUMNS",
+      null, false, getCallOptions());
+    try (FlightStream stream = flightSqlClient.getStream(flightInfo.getEndpoints().get(0).getTicket(),
+      getCallOptions())) {
+      Assert.assertTrue(stream.next());
+      VectorSchemaRoot root = stream.getRoot();
+
+      Assert.assertEquals(root.getRowCount(), 1);
+    }
+  }
+
+  @Test
+  public void testGetTablesFilteringByTableTypePattern() throws Exception {
+    FlightSqlClient flightSqlClient = getFlightClientWrapper().getSqlClient();
+    FlightInfo flightInfo = flightSqlClient.getTables(null, null, null,
+      Collections.singletonList("SYSTEM_TABLE"), false, getCallOptions());
+    try (FlightStream stream = flightSqlClient.getStream(flightInfo.getEndpoints().get(0).getTicket(),
+      getCallOptions())) {
+      Assert.assertTrue(stream.next());
+      VectorSchemaRoot root = stream.getRoot();
+
+      Assert.assertEquals(root.getRowCount(), 28);
+    }
+  }
+
+  @Test
+  public void testGetTablesFilteringByMultiTableTypes() throws Exception {
+    FlightSqlClient flightSqlClient = getFlightClientWrapper().getSqlClient();
+    FlightInfo flightInfo = flightSqlClient.getTables(null, null, null,
+      ImmutableList.of("TABLE", "VIEW"), false, getCallOptions());
+    try (FlightStream stream = flightSqlClient.getStream(flightInfo.getEndpoints().get(0).getTicket(),
+      getCallOptions())) {
+      Assert.assertTrue(stream.next());
+      VectorSchemaRoot root = stream.getRoot();
+
+      Assert.assertEquals(root.getRowCount(), 0);
+    }
+  }
+
+  @Test
+  public void testGetTablesTypes() throws Exception {
+    FlightSqlClient flightSqlClient = getFlightClientWrapper().getSqlClient();
+    FlightInfo flightInfo = flightSqlClient.getTableTypes(getCallOptions());
+    try (
+      FlightStream stream = flightSqlClient.getStream(flightInfo.getEndpoints().get(0).getTicket(), getCallOptions())) {
+      Assert.assertTrue(stream.next());
+      VectorSchemaRoot root = stream.getRoot();
+
+      final ImmutableList<TableType> tableTypes =
+        ImmutableList.of(TableType.TABLE, TableType.SYSTEM_TABLE, TableType.VIEW);
+      final int counter = tableTypes.size();
+
+      final IntStream range = IntStream.range(0, counter);
+
+      Assert.assertEquals(root.getRowCount(), counter);
+      range.forEach(i -> {
+        Assert.assertEquals(root.getVector(0).getObject(i).toString(), tableTypes.get(i).toString());
+      });
+    }
+  }
+
+  private void testGetSchemas(String catalog, String schemaPattern, boolean expectNonEmptyResult) throws Exception {
+    FlightSqlClient flightSqlClient = getFlightClientWrapper().getSqlClient();
+    FlightInfo flightInfo = flightSqlClient.getSchemas(catalog, schemaPattern, getCallOptions());
```

Comment:
final.

Also this is coming up a lot. Should it go in the @Before method and promote these to member variables?

Reply:
Fixed